cedar-drt/fuzz/src/schemas.rs

```diff
@@ -83,171 +85,301 @@ fn remove_trivial_empty_namespace<N>(schema: &mut json_schema::Fragment<N>) {
+                    action_type_equivalence(name.as_ref(), lhs_action, rhs_action)
+                })
+                .fold(Ok(()), Result::and)
+        }
     }
 }
 
-type EntityData<N> = HashMap<UnreservedId, EntityType<N>>;
+impl<K: Eq + Hash + Display, V: Equiv> Equiv for HashMap<K, V> {
+    fn equiv(lhs: &HashMap<K, V>, rhs: &HashMap<K, V>) -> Result<(), String> {
+        if lhs.len() == rhs.len() {
+            let errors = lhs
+                .iter()
+                .filter_map(|(k, lhs_v)| match rhs.get(k) {
+                    Some(rhs_v) => Equiv::equiv(lhs_v, rhs_v).err(),
+                    None => Some(format!("`{k}` missing from rhs")),
+                })
+                .collect::<Vec<_>>();
+            if errors.is_empty() {
+                Ok(())
+            } else {
+                Err(format!(
+                    "Found the following mismatches: {}",
+                    errors.into_iter().join("\n")
+                ))
+            }
+        } else {
+            let lhs_keys: HashSet<_> = lhs.keys().collect();
+            let rhs_keys: HashSet<_> = rhs.keys().collect();
+            let missing_keys = lhs_keys.symmetric_difference(&rhs_keys).join(", ");
+            Err(format!("Missing keys: {missing_keys}"))
+        }
+    }
+}
 
-fn entity_types_equivalence<N: Clone + PartialEq + Debug + Display + TypeName + Ord>(
-    lhs: EntityData<N>,
-    rhs: EntityData<N>,
-) -> Result<(), String> {
-    if lhs.len() == rhs.len() {
-        let errors = lhs
-            .into_iter()
-            .filter_map(|lhs| entity_type_equivalence(lhs, &rhs).err())
-            .collect::<Vec<_>>();
-        if errors.is_empty() {
-            Ok(())
+impl<K: Eq + Ord + Display, V: Equiv> Equiv for BTreeMap<K, V> {
+    fn equiv(lhs: &BTreeMap<K, V>, rhs: &BTreeMap<K, V>) -> Result<(), String> {
+        if lhs.len() == rhs.len() {
+            let errors = lhs
+                .iter()
+                .filter_map(|(k, lhs_v)| match rhs.get(k) {
+                    Some(rhs_v) => Equiv::equiv(lhs_v, rhs_v).err(),
+                    None => Some(format!("`{k}` missing from rhs")),
+                })
+                .collect::<Vec<_>>();
+            if errors.is_empty() {
+                Ok(())
+            } else {
+                Err(format!(
+                    "Found the following mismatches: {}",
+                    errors.into_iter().join("\n")
+                ))
+            }
         } else {
+            let lhs_keys: BTreeSet<_> = lhs.keys().collect();
+            let rhs_keys: BTreeSet<_> = rhs.keys().collect();
+            let missing_keys = lhs_keys.symmetric_difference(&rhs_keys).join(", ");
+            Err(format!("Missing keys: {missing_keys}"))
+        }
+    }
+}
+
+impl<N: Clone + PartialEq + Debug + Display + TypeName + Ord> Equiv for EntityType<N> {
+    fn equiv(lhs: &Self, rhs: &Self) -> Result<(), String> {
+        if !vector_equiv(&lhs.member_of_types, &rhs.member_of_types) {
```

Comment:
Any reason `vector_equiv` isn't also an implementation of `Equiv`?

Reply:
It actually does set equivalence, and I suppose it's unclear whether we _always_ want set equivalence whenever we're applying `Equiv` to vectors?

It would probably be reasonable to make it the canonical `Equiv` implementation though, and assume that we never have ordered lists in the schema format (or anything else we're `Equiv`ing)

Reply:
Removed `vector_equiv` entirely by collecting into hashset/btreeset first. Makes more sense to do set equivalence that way instead of sorting anyway.